error.go

```diff
@@ -250,6 +260,30 @@ func (e *ErrorResponse) ResponseStatus() int { return e.Status }
 // Token is the unique error occurrence identifier.
 func (e *ErrorResponse) Token() string { return e.ID }
 
+// Merge handles the logic of combining two errors together.
+func (e *ErrorResponse) Merge(other error) error {
+	o := asErrorResponse(other)
+	switch {
+	case e.Status == 500 || o.Status == 500:
+		if e.Status != 500 {
+			e.Status = 500
+			e.Code = "internal_error"
+		}
+	case e.Status != o.Status || e.Code != o.Code:
+		e.Status = 400
+		e.Code = "bad_request"
+	}
+	e.Detail = e.Detail + "; " + o.Detail
+
+	if e.Meta == nil && len(o.Meta) > 0 {
+		e.Meta = make(map[string]interface{})
+	}
+	for k, v := range o.Meta {
+		e.Meta[k] = v
+	}
+	return e
+}
+
 // MergeErrors updates an error by merging another into it. It first converts other into a
```

Comment:
This comment should be moved to the `ErrorResponse.Merge` method since the logic may now be different. The new comment should explain that the algo first checks whether err is mergeable and is so merges o into it then whether o is mergeable and finally creates a mergeable error with status code 500 if none is.